src/Common/src/System/Net/ContextAwareResult.cs

```diff
@@ -192,19 +210,29 @@ internal WindowsIdentity Identity
                 }
 
                 // Make sure the identity was requested.
-                GlobalLog.Assert((_flags & StateFlags.CaptureIdentity) != 0, "ContextAwareResult#{0}::Identity|No identity captured - specify captureIdentity.", LoggingHash.HashString(this));
+                if ((_flags & StateFlags.CaptureIdentity) == 0 && GlobalLog.IsEnabled)
+                {
+                    GlobalLog.AssertFormat("ContextAwareResult#{0}::Identity|No identity captured - specify captureIdentity.", LoggingHash.HashString(this));
+                }
 
                 // Just use the lock to block.  We might be on the thread that owns the lock which is great, it means we
                 // don't need an identity anyway.
                 if ((_flags & StateFlags.PostBlockFinished) == 0)
                 {
-                    GlobalLog.Assert(_lock != null, "ContextAwareResult#{0}::Identity|Must lock (StartPostingAsyncOp()) { ... FinishPostingAsyncOp(); } when calling Identity (unless it's only called after FinishPostingAsyncOp).", LoggingHash.HashString(this));
+                    if (_lock == null)
+                    {
+                        GlobalLog.AssertFormat("ContextAwareResult#{0}::Identity|Must lock (StartPostingAsyncOp()) { ... FinishPostingAsyncOp(); } when calling Identity (unless it's only called after FinishPostingAsyncOp).", LoggingHash.HashString(this));
+                    }
                     lock (_lock) { }
                 }
 
-                GlobalLog.Assert(!InternalPeekCompleted || (_flags & StateFlags.ThreadSafeContextCopy) != 0, "ContextAwareResult#{0}::Identity|Result became completed during call.", LoggingHash.HashString(this));
                 if (InternalPeekCompleted)
                 {
+                    if ((_flags & StateFlags.ThreadSafeContextCopy) == 0)
+                    {
+                        GlobalLog.AssertFormat("ContextAwareResult#{0}::Identity|Result became completed during call.", LoggingHash.HashString(this));
```

Comment:
Why is this call to `GlobalLog.AssertFormat` not protected by a call to `Globallog.IsEnabled` but other calls to `GlobalLog.AssertFormat` are protected?